tests/unit/test_pareto.py

```diff
@@ -0,0 +1,50 @@
+# Copyright 2020 The Trieste Contributors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import numpy.testing as npt
+import pytest
+import tensorflow as tf
+
+from trieste.data import Dataset
+from trieste.utils.pareto import non_dominated_sort
+
+
+@pytest.mark.parametrize(
```

Comment:
parametrize is only necessary if you're running the same test for multiple values

Reply:
that said, it might help to have a test with a couple of duplicate points

Reply:
duplicate points? just copy the current param setting?

Reply:
you could copy the current test, but include one of the points twice, e.g. 
```
...
                        [0.9649, 0.9157],
                        [0.1576, 0.7922],
                        [0.1576, 0.7922],
                        [0.9706, 0.9595],
...
```

Reply:
and also a test where points agree only along one dim. On reflection, this could all be in one test with the lines
```
[0.1576, 0.7922],
[0.1576, 0.7922],
[0.9706, 0.9595],
[0.9706, 0.5000],
```

Reply:
Apparently, the current implementation couldn't handle duplicate points. Is it necessary? usually, we won't update the observation when the sampler/BO returns duplicate points. we can of course put some duplicate filter on `non_dominated`, but wouldn't it make the function slower?

Reply:
> usually, we won't update the observations when the sampler/BO returns duplicate point(s)

I don't quite understand. I'm not aware of any such filtering at the moment. We just add whatever comes out of the observer to the current data. Two different query points could have the same output could they not? I realise it's highly unlikely for random sampling in a continuous space, but since it's not unphysical, it would seem odd to me to prohibit it.

Have I understood you right?

Reply:
Yes, you understand me right. seems that it's just my assumption that we have the filtering somewhere. 
I see. yes you're right, in some cases we may get same output on different points.

Reply:
seems it's not so straightforward to make a duplicate filter in TensorFlow. in NumPy we can easily use `np.unique` which has an axis option on it, while in Tensorflow tf.unique only work on a 1D tensor. I found this GitHub issue regarding this problem [link](https://github.com/tensorflow/tensorflow/issues/15644) and also someone mentions a [workaround](https://github.com/tensorflow/tensorflow/issues/15644#issuecomment-407222423) which seems again -back to my question- wouldn't it make the function slower? 
how's the chance we got duplicate points?. if we keep the duplicate points what might go wrong?

note: maybe you have an approach that is more efficient and elegant than the above-mentioned workaround?

Reply:
i spoke with victor. If the error that's raised when there are duplicate points is vaguely sensible, we could just document that, and add a test that it raises for duplicate points.

As for points that are the same in some but not all dimensions, does that work?

Reply:
> As for points that are the same in some but not all dimensions, does that work?

just write a test for it, and it fails. might be good to discuss this tomorrow.